## Marketing Strategy Analysis

#### Data import and cleaning

In [91]:
import pandas as pd
import numpy as np
import seaborn as sns
import scipy.stats

In [92]:
data = pd.read_csv(r'E:\Diego\Data_Analyst\Projects\Optimize Marketing Campaigns (Python)\Campaign-Data.csv')

In [93]:
data.head(10)

Client ID      Client Type  Number of Customers  Montly Target  Zip Code  \
0  ID-987275  Medium Facility                 2800            125      1003   
1  ID-987275  Medium Facility                 2800            125      1003   
2  ID-987275  Medium Facility                 2800            125      1003   
3  ID-987275  Medium Facility                 2800            125      1003   
4  ID-987275  Medium Facility                 2800            125      1003   
5  ID-987275  Medium Facility                 2800            125      1003   
6  ID-987275  Medium Facility                 2800            125      1003   
7  ID-987275  Medium Facility                 2800            125      1003   
8  ID-987275  Medium Facility                 2800            125      1003   
9  ID-987275  Medium Facility                 2800            125      1003   

  Calendardate  Amount Collected  Unit Sold  Campaign (Email)  \
0   16-01-2014                 0          0               0.0   
1   16-02-2014           3409460         24               0.0   
2   18-03-2014          10228384         75               0.0   
3   18-04-2014          17047304        123               0.0   
4   19-05-2014          23866224        171               0.0   
5   16-06-2014          27275684        198               0.0   
6   17-07-2014          17047304        123               0.0   
7   16-08-2014          17047304        123               0.0   
8   16-09-2014          17047304        123          383800.8   
9   16-10-2014          27275684        198               0.0   

   Campaign (Flyer)  Campaign (Phone)  Sales Contact 1  Sales Contact 2  \
0               0.0               0.0              0.0              0.0   
1               0.0               0.0              0.0              0.0   
2               0.0               0.0              0.0              0.0   
3               0.0               0.0              0.0        3547500.0   
4               0.0               0.0              0.0              0.0   
5               0.0               0.0              0.0        4515000.0   
6               0.0               0.0              0.0        4515000.0   
7          696213.0               0.0              0.0        3547500.0   
8         1290412.8               0.0              0.0        2580000.0   
9          664917.6               0.0              0.0        2580000.0   

   Sales Contact 3  Sales Contact 4  Sales Contact 5 Number of Competition  
0              0.0              0.0              0.0                   Low  
1              0.0              0.0         322500.0                   Low  
2              0.0              0.0              0.0                   Low  
3        1290000.0              0.0              0.0                   Low  
4              0.0              0.0              0.0                   Low  
5         322500.0              0.0              0.0                   Low  
6        2580000.0              0.0              0.0                   Low  
7        3547500.0              0.0              0.0                   Low  
8        4192500.0              0.0              0.0                   Low  
9        4515000.0              0.0              0.0                   Low

In [94]:
df = data

In [95]:
df.columns.unique()

Index(['Client ID', 'Client Type', 'Number of Customers', 'Montly Target',
       'Zip Code', 'Calendardate', 'Amount Collected', 'Unit Sold',
       'Campaign (Email)', 'Campaign (Flyer)', 'Campaign (Phone)',
       'Sales Contact 1', 'Sales Contact 2', 'Sales Contact 3',
       'Sales Contact 4', 'Sales Contact 5', 'Number of Competition'],
      dtype='object')

In [96]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2976 entries, 0 to 2975
Data columns (total 17 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Client ID              2976 non-null   object 
 1   Client Type            2976 non-null   object 
 2   Number of Customers    2976 non-null   int64  
 3   Montly Target          2976 non-null   int64  
 4   Zip Code               2976 non-null   int64  
 5   Calendardate           2976 non-null   object 
 6   Amount Collected       2976 non-null   int64  
 7   Unit Sold              2976 non-null   int64  
 8   Campaign (Email)       2976 non-null   float64
 9   Campaign (Flyer)       2976 non-null   float64
 10  Campaign (Phone)       2976 non-null   float64
 11  Sales Contact 1        2976 non-null   float64
 12  Sales Contact 2        2976 non-null   float64
 13  Sales Contact 3        2976 non-null   float64
 14  Sales Contact 4        2976 non-null   float64
 15  Sale

In [97]:
#Formatting calendardate column as a date and splitting it
df['Calendardate'] = pd.to_datetime(data['Calendardate'])
df['Calendar_Month'] = df['Calendardate'].dt.month
df['Calendar_Year'] = df['Calendardate'].dt.year

C:\Users\DIEGO\AppData\Local\Temp\ipykernel_12372\4073020052.py:2: UserWarning: Parsing dates in %d-%m-%Y format when dayfirst=False (the default) was specified. Pass `dayfirst=True` or specify a format to silence this warning.
  df['Calendardate'] = pd.to_datetime(data['Calendardate'])


In [98]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2976 entries, 0 to 2975
Data columns (total 19 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   Client ID              2976 non-null   object        
 1   Client Type            2976 non-null   object        
 2   Number of Customers    2976 non-null   int64         
 3   Montly Target          2976 non-null   int64         
 4   Zip Code               2976 non-null   int64         
 5   Calendardate           2976 non-null   datetime64[ns]
 6   Amount Collected       2976 non-null   int64         
 7   Unit Sold              2976 non-null   int64         
 8   Campaign (Email)       2976 non-null   float64       
 9   Campaign (Flyer)       2976 non-null   float64       
 10  Campaign (Phone)       2976 non-null   float64       
 11  Sales Contact 1        2976 non-null   float64       
 12  Sales Contact 2        2976 non-null   float64       
 13  Sal

In [99]:
df.shape

(2976, 19)

### Exploratory Data Analysis

#### Understanding distributions

In [100]:
df['Client Type'].value_counts(normalize=True)

Client Type
Large Facility      0.459677
Small Facility      0.282258
Medium Facility     0.169355
Private Facility    0.088710
Name: proportion, dtype: float64

In [101]:
pd.crosstab(df['Number of Competition'],df['Client Type'], margins=True, normalize='columns')

Client Type            Large Facility   Medium Facility  Private Facility  \
Number of Competition                                                       
High                          0.166667         0.166667          0.166667   
Low                           0.833333         0.833333          0.833333   

Client Type            Small Facility        All  
Number of Competition                             
High                          0.166667  0.166667  
Low                           0.833333  0.833333

In [102]:
df.groupby('Number of Competition').mean(numeric_only=True)

Number of Customers  Montly Target  Zip Code  \
Number of Competition                                                 
High                           1456.935484      75.080645    1003.0   
Low                            1456.935484      75.080645    1003.0   

                       Amount Collected   Unit Sold  Campaign (Email)  \
Number of Competition                                                   
High                       2.974789e+07  213.127016     105398.938508   
Low                        1.445570e+07  103.132258     150862.165766   

                       Campaign (Flyer)  Campaign (Phone)  Sales Contact 1  \
Number of Competition                                                        
High                      994046.717540      45198.036895    146945.564516   
Low                       623692.979839      26693.304194    128219.758065   

                       Sales Contact 2  Sales Contact 3  Sales Contact 4  \
Number of Competition                                                      
High                      2.685333e+06     1.786754e+06     72172.379032   
Low                       1.890916e+06     1.883634e+06     70481.854839   

                       Sales Contact 5  Calendar_Month  Calendar_Year  
Number of Competition                                                  
High                       8452.620968            10.5         2015.0  
Low                       15864.919355             5.7         2014.4

In [103]:
df.groupby('Client Type').mean(numeric_only=True)

Number of Customers  Montly Target  Zip Code  \
Client Type                                                      
Large Facility            1380.842105      71.578947    1003.0   
Medium Facility           3940.761905     202.857143    1003.0   
Private Facility           400.727273      20.454545    1003.0   
Small Facility             422.514286      21.285714    1003.0   

                  Amount Collected   Unit Sold  Campaign (Email)  \
Client Type                                                        
Large Facility        1.999880e+07  143.098684     142273.609649   
Medium Facility       4.075997e+07  290.583333     437217.097817   
Private Facility      5.030246e+06   35.784091       5183.715152   
Small Facility        1.637759e+06   11.689286      11975.986310   

                  Campaign (Flyer)  Campaign (Phone)  Sales Contact 1  \
Client Type                                                             
Large Facility        8.192056e+05      45595.436623    133667.763158   
Medium Facility       1.552603e+06      49176.847619    398645.833333   
Private Facility      2.272919e+05       5522.470455      1221.590909   
Small Facility        9.120875e+04          0.000000      8062.500000   

                  Sales Contact 2  Sales Contact 3  Sales Contact 4  \
Client Type                                                           
Large Facility       2.034013e+06     2.017039e+06    119287.280702   
Medium Facility      4.822783e+06     4.698646e+06     85104.166667   
Private Facility     6.376705e+05     4.434375e+05      3664.772727   
Small Facility       7.617143e+05     3.727946e+05      4223.214286   

                  Sales Contact 5  Calendar_Month  Calendar_Year  
Client Type                                                       
Large Facility       16266.447368             6.5         2014.5  
Medium Facility      33273.809524             6.5         2014.5  
Private Facility     12215.909091             6.5         2014.5  
Small Facility        1535.714286             6.5         2014.5

In [104]:
data.corr(numeric_only=True)[['Amount Collected']]

Amount Collected
Number of Customers          0.607496
Montly Target                0.608204
Zip Code                          NaN
Amount Collected             1.000000
Unit Sold                    0.997515
Campaign (Email)             0.248235
Campaign (Flyer)             0.444337
Campaign (Phone)             0.034858
Sales Contact 1              0.277478
Sales Contact 2              0.552112
Sales Contact 3              0.357887
Sales Contact 4              0.236165
Sales Contact 5              0.095795
Calendar_Month               0.139425
Calendar_Year                0.286194

#### Correlation Analysis

In [105]:
cm = sns.light_palette('orange', as_cmap=True)
correlation_analysis = pd.DataFrame(df[['Amount Collected', 'Campaign (Email)', 'Campaign (Flyer)', 'Campaign (Phone)', 'Sales Contact 1', 'Sales Contact 2', 'Sales Contact 3',
                                       'Sales Contact 4', 'Sales Contact 5']].corr()['Amount Collected']).reset_index()
correlation_analysis.columns=['Variable', 'Correlation']
correlation_analysis=correlation_analysis[correlation_analysis['Variable']!='Amount Collected']
correlation_analysis=correlation_analysis.sort_values('Correlation', ascending=False)
correlation_analysis.style.background_gradient(cmap=cm)

In [106]:
cm = sns.light_palette('orange', as_cmap=True)
correlation_analysis = pd.DataFrame(df.groupby('Client Type')[['Amount Collected', 'Campaign (Email)', 'Campaign (Flyer)', 'Campaign (Phone)', 'Sales Contact 1', 'Sales Contact 2', 'Sales Contact 3',
                                       'Sales Contact 4', 'Sales Contact 5']].corr()['Amount Collected']).reset_index()
correlation_analysis=correlation_analysis.sort_values(['Client Type', 'Amount Collected'], ascending=False)
correlation_analysis.columns=['Account Type', 'Variable Impact', 'Impact']
correlation_analysis=correlation_analysis[correlation_analysis['Variable Impact']!='Amount Collected'].reset_index(drop=True)
correlation_analysis.style.background_gradient(cmap=cm)

As we can see sales contact 2 has a high impact on sales done in large facilities, medium facilities and private facilities. While phone campaigns seem to be the least effective for most facilities

### Regression Analysis (Market Sales and Strategies)

In [107]:
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [108]:
df.columns=[mystring.replace(' ', '_') for mystring in df.columns]
df.columns=[mystring.replace('(', '') for mystring in df.columns]
df.columns=[mystring.replace(')', '') for mystring in df.columns]
results = smf.ols('Amount_Collected ~ Campaign_Email+Campaign_Flyer+Campaign_Phone+Sales_Contact_1+Sales_Contact_2+Sales_Contact_3+Sales_Contact_4+Sales_Contact_5',data=df).fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:       Amount_Collected   R-squared:                       0.480
Model:                            OLS   Adj. R-squared:                  0.478
Method:                 Least Squares   F-statistic:                     342.1
Date:                Sat, 01 Mar 2025   Prob (F-statistic):               0.00
Time:                        14:56:20   Log-Likelihood:                -54512.
No. Observations:                2976   AIC:                         1.090e+05
Df Residuals:                    2967   BIC:                         1.091e+05
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
Intercept        1.481e+06   5.12e+05     

In [109]:
coef = pd.read_html(results.summary().tables[1].as_html(), header=0, index_col=0)[0]
coef

C:\Users\DIEGO\AppData\Local\Temp\ipykernel_12372\1897702063.py:1: FutureWarning: Passing literal html to 'read_html' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  coef = pd.read_html(results.summary().tables[1].as_html(), header=0, index_col=0)[0]


coef     std err       t  P>|t|      [0.025  \
Intercept        1.481000e+06  512000.000   2.891  0.004  477000.000   
Campaign_Email   7.932000e-01       0.597   1.329  0.184      -0.377   
Campaign_Flyer   3.337600e+00       0.260  12.831  0.000       2.828   
Campaign_Phone   7.340000e-02       1.053   0.070  0.944      -1.991   
Sales_Contact_1  4.236800e+00       0.415  10.207  0.000       3.423   
Sales_Contact_2  3.638200e+00       0.129  28.155  0.000       3.385   
Sales_Contact_3  2.343200e+00       0.131  17.925  0.000       2.087   
Sales_Contact_4  1.094780e+01       1.060  10.331  0.000       8.870   
Sales_Contact_5  3.507800e+00       4.549   0.771  0.441      -5.412   

                      0.975]  
Intercept        2490000.000  
Campaign_Email         1.963  
Campaign_Flyer         3.848  
Campaign_Phone         2.137  
Sales_Contact_1        5.051  
Sales_Contact_2        3.892  
Sales_Contact_3        2.600  
Sales_Contact_4       13.026  
Sales_Contact_5       12.428

In [110]:
coef = coef.reset_index()
coef = coef[coef['P>|t|']<0.05][['index', 'coef']]
coef

index          coef
0        Intercept  1.481000e+06
2   Campaign_Flyer  3.337600e+00
4  Sales_Contact_1  4.236800e+00
5  Sales_Contact_2  3.638200e+00
6  Sales_Contact_3  2.343200e+00
7  Sales_Contact_4  1.094780e+01

### Regression Analysis Broken up by Account Types

In [122]:
summary = pd.DataFrame()
for acctype in list(set(list(df['Client_Type']))):
    print(acctype)
    temp = df[df['Client_Type']==acctype].copy()
    results = smf.ols('Amount_Collected ~ Campaign_Email+Campaign_Flyer+Campaign_Phone+Sales_Contact_1+Sales_Contact_2+Sales_Contact_3+Sales_Contact_4+Sales_Contact_5', data=temp).fit()
    coef = pd.read_html(results.summary().tables[1].as_html(), header=0, index_col=0)[0].reset_index()
    coef = coef[coef['P>|t|']<0.05][['index', 'coef']]
    coef.columns = ['Variable', 'Coefficient (Impact)']
    coef['Account Type'] = acctype
    coef = coef.sort_values('Coefficient (Impact)', ascending=False)
    coef = coef[coef['Variable']!= 'Intercept']
    summary = pd.concat([summary, coef])
    print(results.summary())

Large Facility 
                            OLS Regression Results                            
Dep. Variable:       Amount_Collected   R-squared:                       0.371
Model:                            OLS   Adj. R-squared:                  0.367
Method:                 Least Squares   F-statistic:                     100.2
Date:                Sat, 01 Mar 2025   Prob (F-statistic):          4.64e-131
Time:                        15:08:16   Log-Likelihood:                -25075.
No. Observations:                1368   AIC:                         5.017e+04
Df Residuals:                    1359   BIC:                         5.021e+04
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
Intercept        2.812e+06

C:\Users\DIEGO\AppData\Local\Temp\ipykernel_12372\885186593.py:6: FutureWarning: Passing literal html to 'read_html' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  coef = pd.read_html(results.summary().tables[1].as_html(), header=0, index_col=0)[0].reset_index()
C:\Users\DIEGO\AppData\Local\Temp\ipykernel_12372\885186593.py:6: FutureWarning: Passing literal html to 'read_html' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  coef = pd.read_html(results.summary().tables[1].as_html(), header=0, index_col=0)[0].reset_index()
C:\Users\DIEGO\AppData\Local\Temp\ipykernel_12372\885186593.py:6: FutureWarning: Passing literal html to 'read_html' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  coef = pd.read_html(results.summary().tables[1].as_html(), header=0, index_col=0)[0].reset_

### Final Recommendations

Using the table below, we can use the coefficient to see how much return we can derive from each dollar we spend. Each account type has different methods that are effective to them.

#### Medium Facility

In [123]:
summary[summary['Account Type']=='Medium Facility']

Variable  Coefficient (Impact)     Account Type
2   Campaign_Flyer                4.1059  Medium Facility
5  Sales_Contact_2                3.5778  Medium Facility
4  Sales_Contact_1                3.1365  Medium Facility
6  Sales_Contact_3                2.1174  Medium Facility

In [113]:
def format(x):
    return '${:.1f}'.format(x)

In [133]:
summary.rename(columns={'Coefficient (Impact)':'Return On Investment'}, inplace=True)
summary['Return On Investment'] = summary['Return On Investment'].apply(lambda x: round(x,1))

In [134]:
summary['Return On Investment'] = summary['Return On Investment'].apply(format)

In [135]:
summary[summary['Account Type']=='Medium Facility']

Variable Return On Investment     Account Type
2   Campaign_Flyer                 $4.1  Medium Facility
5  Sales_Contact_2                 $3.6  Medium Facility
4  Sales_Contact_1                 $3.1  Medium Facility
6  Sales_Contact_3                 $2.1  Medium Facility

For each dollar spend on flyer campaigns, the average return will be $4.1 dollars. Sales contact 1, 2 and 3 are also highly effective.
Other strategies used show no impact so could be dropped to save costs.

#### Private Facility

In the case of the private facilities, the only succesful campaign so far has been sales contact 2, all the others haven't make an impact in increasing the sales.

In [138]:
summary

Variable Return On Investment      Account Type
4  Sales_Contact_1                $11.7   Large Facility 
7  Sales_Contact_4                $10.6   Large Facility 
5  Sales_Contact_2                 $4.0   Large Facility 
2   Campaign_Flyer                 $2.7   Large Facility 
6  Sales_Contact_3                 $2.0   Large Facility 
3   Campaign_Phone                $-3.5   Large Facility 
2   Campaign_Flyer                 $4.1   Medium Facility
5  Sales_Contact_2                 $3.6   Medium Facility
4  Sales_Contact_1                 $3.1   Medium Facility
6  Sales_Contact_3                 $2.1   Medium Facility
5  Sales_Contact_2                 $0.8   Small Facility 
3   Campaign_Phone                $-0.0   Small Facility 
5  Sales_Contact_2                 $6.6  Private Facility

#### Large Facilities

Large Facilities have the best 2 campaigns, sales contact 1 and 4, they have return on investment of 11.7 and 10.6 respectively for each dollar invested. The advise would be to increase the budget for those campaigns, and drop the phone campaign which is not even bringing any positive returns.